File: EmployeeTestApi/Data/Models/EmployeesFilterModel.cs
using System;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.ModelBinding;

namespace EmployeeTestApi.Data.Models
{
    public class EmployeesFilterModel
    {
        public string EmployeeFullName { get; set; }
        public string DepartmentName { get; set; }
        public string SalarySearchStr { get; set; }
        public string DateOfBirth { get; set; }
        public string DateOfEmployment { get; set; }
        public string SortColumnName { get; set; }
        public bool AscDirection { get; set; }
    }
}

File: employeeClient/src/app/_models/employees-filter-model.ts
export class EmployeesFilterModel {
    departmentName: string;
    employeeFullName: string;
    dateOfBirth: string;
    dateOfEmployment: string;
    salarySearchStr: string;
    sortColumnName: string;
    ascDirection: boolean;

    constructor() {
        this.departmentName = '';
        this.employeeFullName = '';
        this.salarySearchStr ='';
        this.dateOfBirth ='';
        this.dateOfEmployment ='';
        this.sortColumnName = 'FullName';
        this.ascDirection = true;
    }
}
